src/tests/btree_helpers/btree_test_helper.hpp

```diff
@@ -100,60 +100,63 @@ struct BtreeTestHelper {
-            auto test_count = n_fibers;
-
-            for (std::size_t i = 0; i < n_fibers; ++i) {
-                const auto start_range = i * chunk_size;
-                const auto end_range = start_range + ((i == n_fibers - 1) ? last_chunk_size : chunk_size);
-                auto fiber_id = i;
-                iomanager.run_on_forget(
-                    m_fibers[i], [this, start_range, end_range, &test_count, fiber_id, preload_size]() {
-                        double progress_interval =
-                            (double)(end_range - start_range) / 20; // 5% of the total number of iterations
-                        double progress_thresh = progress_interval; // threshold for progress interval
-                        double elapsed_time, progress_percent, last_progress_time = 0;
-                        auto m_start_time = Clock::now();
-
-                        for (uint32_t i = start_range; i < end_range; i++) {
-                            put(i, btree_put_type::INSERT);
-                            if (fiber_id == 0) {
-                                elapsed_time = get_elapsed_time_sec(m_start_time);
-                                progress_percent = (double)(i - start_range) / (end_range - start_range) * 100;
-
-                                // check progress every 5% of the total number of iterations or every 30 seconds
-                                bool print_time = false;
-                                if (i >= progress_thresh) {
-                                    progress_thresh += progress_interval;
-                                    print_time = true;
-                                }
-                                if (elapsed_time - last_progress_time > 30) {
-                                    last_progress_time = elapsed_time;
-                                    print_time = true;
-                                }
-                                if (print_time) {
-                                    LOGINFO("Progress: iterations completed ({:.2f}%)- Elapsed time: {:.0f} seconds- "
-                                            "populated entries: {} ({:.2f}%)",
-                                            progress_percent, elapsed_time, m_shadow_map.size(),
-                                            m_shadow_map.size() * 100.0 / preload_size);
-                                }
-                            }
+        if (preload_size == 0) {
+            LOGINFO("Preload Skipped");
+            return;
+        }
+
+        const auto n_fibers = std::min(preload_size, (uint32_t)m_fibers.size());
+        const auto chunk_size = preload_size / n_fibers;
+        const auto last_chunk_size = preload_size % chunk_size ?: chunk_size;
+        auto test_count = n_fibers;
+
+        for (std::size_t i = 0; i < n_fibers; ++i) {
+            const auto start_range = i * chunk_size;
+            const auto end_range = start_range + ((i == n_fibers - 1) ? last_chunk_size : chunk_size) - 1;
```

Comment:
There was a bug in getting end_range, which was exposed while testing with smaller number of preloads. Fixed that.